lib/Exporter/Writer/KnpSnappyWriter.php

```diff
@@ -0,0 +1,181 @@
+            throw new \RuntimeException('Unable to find wkhtmltopdf binary');
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function open()
+    {
+        $this->tmpFile = tmpfile();
+        fwrite($this->tmpFile, '<html><head><meta http-equiv="Content-Type" content="text/html; charset=utf-8" /><meta name=ProgId content=Pdf.Document><meta name=Generator content="https://github.com/sonata-project/exporter"><link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.5/css/bootstrap.min.css" crossorigin="anonymous"></head><body>');
+
+        if (isset($this->snappyOptions['header-html'])) {
+            fwrite($this->tmpFile, $this->snappyOptions['header-html']);
+            unset($this->snappyOptions['header-html']);
+        }
+        fwrite($this->tmpFile, '<table class="table ');
+        if (true === $this->showBorders) {
+            fwrite($this->tmpFile, 'table-bordered');
+        }
+
+        fwrite($this->tmpFile, '" >');
+    }
+
+    /**
+     * @param $data
+     *
+     * @return array mixed
+     */
+    protected function init($data)
+    {
+        if ($this->position > 0) {
+            return;
+        }
+
+        if (true === $this->showHeaders) {
+            fwrite($this->tmpFile, '<tr>');
+            foreach ($data as $header => $value) {
+                fwrite($this->tmpFile, sprintf('<th>%s</th>', $header));
+            }
+            fwrite($this->tmpFile, '</tr>');
+            ++$this->position;
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function write(array $data)
+    {
+        $this->init($data);
+
+        fwrite($this->tmpFile, '<tr>');
+        foreach ($data as $value) {
+            fwrite($this->tmpFile, sprintf('<td>%s</td>', $value));
+        }
+        fwrite($this->tmpFile, '</tr>');
+
+        ++$this->position;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function close()
+    {
+        $tmpFileMetaData = stream_get_meta_data($this->tmpFile);
+        $tmpFilePath = $tmpFileMetaData['uri'];
+        fwrite($this->tmpFile, '</table>');
+        if (isset($this->snappyOptions['footer-html'])) {
+            fwrite($this->tmpFile, $this->snappyOptions['footer-html']);
+            unset($this->snappyOptions['footer-html']);
+        }
+        fwrite($this->tmpFile, '</body></html>');
+
+        $this->generate($tmpFilePath);
+        fclose($this->tmpFile);
+    }
+
+    /**
+     * Generates a Pdf document.
+     */
+    protected function generate($tmpFilePath)
+    {
+        $snappy = new Pdf();
+        $snappy->setBinary($this->wkhtmltopdfBinary);
+
+        if (!is_null($this->snappyOptions)) {
+            foreach ($this->snappyOptions as $key => $value) {
+                $snappy->setOption($key, $value);
+            }
+        }
+
+        $html = file_get_contents($tmpFilePath);
+        $output = fopen($this->filename, 'w', false);
+        fwrite($output, $snappy->getOutputFromHtml($html));
```

Comment:
`$snappy->generateFromHtml()` is causing a bug on integrating KnpSnappyWriter in SonataCore/Exporter , because it can't write in `php://output`.
This is why I've replaced it by `getOutputFromHtml()` and manually use `fopen` ,  `fwrite`and `fclose` to create output.
Now it works perfectly. Same on SonataCore integration  (test of compatibility).